### PR TITLE
Replace operator_type with an enum

### PR DESCRIPTION
**Pull request by dekimir**
operator_type is awkward because it's not copyable or assignable. Replace it with a new enum class.

Tests: unit(dev)

### CODE REVIEW

File: cql3/expr/expression.cc [hunk 1]
```diff
@@ -870,6 +885,55 @@ expression replace_column_def(const expression& expr, const column_definition* n
...
+    };
+    const auto found = opmap.find(&op);
+    if (found == opmap.end()) {
+        throw std::logic_error(format("make_column_op: Unrecognized operator_type: {}", op));
+    }
+    return binary_operator{column_value(cdef), found->second, std::move(value)};
+}
+
+std::ostream& operator<<(std::ostream& s, oper_t op) {
+    switch(op) {
```
Comment (avikivity) [inline cql3/expr/expression.cc:889]: space after switch

File: cql3/expr/expression.hh [hunk 2]
```diff
@@ -22,6 +22,7 @@
 #pragma once
 
 #include <fmt/core.h>
+#include <ostream>
```
Comment (avikivity) [inline cql3/expr/expression.hh:25]: I think <iosfwd> is sufficient here.

File: cql3/expr/expression.hh [hunk 3]
```diff
@@ -172,24 +181,46 @@ size_t count_if(const expression& e, Fn f) {
...
+    return (op == oper_t::LT) || (op == oper_t::LTE) || (op == oper_t::GT) || (op == oper_t::GTE);
 }
 
 inline bool has_slice(const expression& e) {
-    return find_atom(e, [] (const binary_operator& o) { return o.op->is_slice(); });
+    return find_atom(e, [] (const binary_operator& bo) { return is_slice(bo.op); });
+}
+
+inline bool is_compare(oper_t op) {
+    switch(op) {
```
Comment (avikivity) [inline cql3/expr/expression.hh:204]: switch space

File: cql3/column_condition.cc [hunk 4]
```diff
@@ -82,21 +83,24 @@ int is_satisfied_by(const cql3::operator_type &op, const abstract_type& cell_typ
         } else {
             rc = cell_type.compare(cell_type.decompose(cell_value), param);
         }
-        if (op == cql3::operator_type::EQ) {
+        switch(op) {
```
Comment (avikivity) [inline cql3/column_condition.cc:86]: space after switch

File: cql3/expr/expression.hh [hunk 5]
```diff
@@ -231,7 +231,9 @@ extern bool is_on_collection(const binary_operator&);
 extern expression replace_column_def(const expression&, const column_definition*);
 
 /// Makes a binary_operator on a column_definition.
-extern expression make_column_op(const column_definition*, const operator_type&, ::shared_ptr<term>);
+inline expression make_column_op(const column_definition* cdef, oper_t op, ::shared_ptr<term> value) {
+    return binary_operator{column_value(cdef), op, std::move(value)};
+}
```
Comment (avikivity) [inline cql3/expr/expression.hh:236]: This is a candidate for removal IMO. The implementation is clearer than the interface and can just be inlined into its callers. Not to interfere with this series though.

Comment (dekimir) [reply cql3/expr/expression.hh:236]: I like the clarity of inlining it, but I hate repetitive typing, and `make_column_op` is shorter than `binary_operator{column_value}`.

File: cql3/expr/expression.hh (hunk 2 above)
Comment (dekimir) [reply cql3/expr/expression.hh:25]: Sorry, I don't understand what you meant by this?

**Comment by dekimir** [comment]
Added space after `switch`, PTAL.  Changelog:
185b4bcd2 Add space after `switch`
f51d24169 Add space after `switch`

File: cql3/column_condition.cc (hunk 4 above)
Comment (dekimir) [reply cql3/column_condition.cc]: Done.

File: cql3/expr/expression.hh (hunk 3 above)
Comment (dekimir) [reply cql3/expr/expression.hh]: Done.

File: cql3/expr/expression.cc (hunk 1 above)
Comment (dekimir) [reply cql3/expr/expression.cc]: Done.

File: cql3/expr/expression.hh (hunk 5 above)
Comment (avikivity) [reply cql3/expr/expression.hh:236]: Perhaps C++ will understand binary_operator{my_cdef, oper_t::EQ, std::move(term)}.

**Comment by avikivity** [comment]
Queued.

File: cql3/expr/expression.hh (hunk 5 above)
Comment (dekimir) [reply cql3/expr/expression.hh:236]: You are correct. Removed in #7099.

**Pull request merged**